Makefile

```diff
@@ -1,92 +1,145 @@
-
-CGC = wine tools/cg/cgc.exe
-
-CXBE = tools/cxbe/cxbe
-VP20COMPILER = tools/vp20compiler/vp20compiler
-FP20COMPILER = tools/fp20compiler/fp20compiler
-
-
-CFLAGS = -target i386-pc-win32 -march=pentium3 \
-    -ffreestanding -nostdlib -fno-builtin -fno-exceptions \
-    -I. -Ixboxrt
-
-SHADERINT = \
-	app/vs.vp \
-	app/ps.fp
-
-SHADEROBJ = \
-	app/vs.inl \
-	app/ps.inl
-
-USB_SRCS = \
-    usb/host/ohci-hcd.c \
-    usb/core/message.c \
-    usb/core/hcd.c \
-    usb/core/hcd-pci.c \
-    usb/core/hub.c \
-    usb/core/usb.c \
-    usb/core/config.c \
-    usb/core/urb.c \
-    usb/core/buffer_simple.c \
-    usb/core/usb-debug.c \
-    usb/sys/BootUSB.c \
-    usb/sys/linuxwrapper.c \
-    usb/sys/xpad.c \
-    usb/sys/xremote.c \
-    usb/sys/usbkey.c \
-    usb/sys/usbmouse.c \
-    usb/misc/misc.c \
-    usb/misc/pci.c \
-    usb/misc/malloc.c
-
-INCLUDES := $(wildcard xboxkrnl/*.h) \
-            $(wildcard xboxrt/*.h) \
-            $(wildcard hal/*.h) \
-            $(wildcard pbkit/*.h) \
-            $(wildcard app/*.h) \
-            $(SHADEROBJ)
-
-SRCS := $(wildcard xboxrt/*.c) \
-        $(wildcard hal/*.c) \
-        $(wildcard pbkit/*.c) \
-        $(wildcard app/*.c) \
-        $(USB_SRCS)
+ifeq ($(NXDK_DIR),)
+NXDK_DIR = $(shell pwd)
+endif
+
+ifeq ($(XBE_TITLE),)
+XBE_TITLE = nxdk_app
+endif
+
+ifeq ($(OUTPUT_DIR),)
+OUTPUT_DIR = bin
+endif
+
+UNAME_S := $(shell uname -s)
+ifeq ($(UNAME_S),Linux)
+LD           = x86_64-w64-mingw32-ld
+CC           = clang
+CXX          = clang++
+endif
+ifeq ($(UNAME_S),Darwin)
+LD           = /usr/local/opt/llvm/bin/lld -flavor link
+CC           = /usr/local/opt/llvm/bin/clang
+CXX          = /usr/local/opt/llvm/bin/clang++
+endif
+
+TARGET       = $(OUTPUT_DIR)/default.xbe
+CGC          = wine $(NXDK_DIR)/tools/cg/cgc.exe
+CXBE         = $(NXDK_DIR)/tools/cxbe/cxbe
+VP20COMPILER = $(NXDK_DIR)/tools/vp20compiler/vp20compiler
+FP20COMPILER = $(NXDK_DIR)/tools/fp20compiler/fp20compiler
+EXTRACT_XISO = $(NXDK_DIR)/tools/extract-xiso/extract-xiso
+TOOLS        = cxbe vp20compiler fp20compiler extract-xiso
+CFLAGS       = -target i386-pc-win32 -march=pentium3 \
+               -ffreestanding -nostdlib -fno-builtin -fno-exceptions \
+               -I$(NXDK_DIR)/lib -I$(NXDK_DIR)/lib/xboxrt \
+               -Wno-ignored-attributes
+
+ifeq ($(DEBUG),y)
+CFLAGS += -g
+endif
+
+include $(NXDK_DIR)/lib/Makefile
 OBJS = $(SRCS:.c=.obj)
 
-bin/default.xbe: app/main.exe
-	mkdir -p bin
-	$(CXBE) -OUT:bin/default.xbe -TITLE:0ldskoo1 app/main.exe
-
-app/main.exe: $(OBJS)
-	$(LLDLINK) -subsystem:windows -dll -out:'$@' -entry:XboxCRT xboxkrnl/libxboxkrnl.lib $(OBJS)
-
-%.obj: %.c ${INCLUDES}
-	$(CC) $(CFLAGS) -c -o '$@' '$<'
-
-app/vs.vp: app/vs.cg
-	$(CGC) -profile vp20 -o '$@' '$<'
-
-app/ps.fp: app/ps.cg
-	$(CGC) -profile fp20 -o '$@' '$<'
-
-%.inl: %.vp
-	$(VP20COMPILER) '$<' > '$@'
-
-%.inl: %.fp
-	$(FP20COMPILER) '$<' > '$@'
+ifneq ($(GEN_XISO),)
+TARGET += $(GEN_XISO)
+endif
+
+V = 0
```

Comment:
I agree with keeping the Makefile simple. I'm ok with ditching distclean if you really want.

That said, the reason I added `$(VE)` and `$(STDOUT_TO_NULL)` is as we keep adding library code the number of files that go into a build shoots up, and when you're compiling it generates heaps of output.

The `$(VE)` bit is nice because if you want to see all the build commands when your build you just run `make V=1`. (Same goes with `$(STDOUT_TO_NULL)` which squelches noisy tools like extract-xiso.) Otherwise, it'll just spit out the filename name of what's currently building. This is pretty common practice nowadays (Linux, coreboot, etc.).

If you feel strongly about it, we can remove this feature.